examples/fabric-admin/device_manager/PairingManager.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <controller/CHIPDeviceController.h>
+#include <controller/CommissioningWindowOpener.h>
+#include <crypto/CHIPCryptoPAL.h>
+
+class CommissioningWindowDelegate
+{
+public:
+    virtual void OnCommissioningWindowOpened(chip::NodeId deviceId, CHIP_ERROR err, chip::SetupPayload payload) = 0;
+    virtual ~CommissioningWindowDelegate()                                                                      = default;
+};
+
+class PairingManager
+{
+public:
+    static PairingManager & Instance()
+    {
+        static PairingManager instance;
+        return instance;
+    }
+
+    CHIP_ERROR Init(chip::Controller::DeviceCommissioner * commissioner);
+
+    CHIP_ERROR OpenCommissioningWindow(chip::NodeId nodeId, chip::EndpointId endpointId, uint16_t commissioningTimeout,
+                                       uint32_t iterations, uint16_t discriminator, const chip::ByteSpan & salt,
+                                       const chip::ByteSpan & verifier);
+
+    void RegisterOpenCommissioningWindowDelegate(CommissioningWindowDelegate * delegate)
+    {
+        ChipLogProgress(NotSpecified, "yujuan: PairingManager::RegisterOpenCommissioningWindowDelegate");
+        mCommissioningWindowDelegate = delegate;
+    }
+
+    void UnregisterOpenCommissioningWindowDelegate() { mCommissioningWindowDelegate = nullptr; }
```

Comment:
How about just having a `SetDelegate` call that accepts null?

Reply:
We need to add other delegate later, change to SetOpenCommissioningWindowDelegate